config/config.go

```diff
@@ -266,6 +279,39 @@ func readJsonConfigFromRepo(repoPath repos.RepoPath) (jsonConfig, error) {
 	return config, nil
 }
 
+// UnmarshalJSON implements the Unmarshaler interface for the jsonParents type, allowing the 'parent' field to be a single
+// struct or array of structs when defined in the config file.
+// @llr REQ-TRAQ-SWL-53
+func (op *jsonParents) UnmarshalJSON(data []byte) error {
+	if len(data) == 0 {
+		return errors.New("no bytes to unmarshal")
+	}
+
+	// Use the first character to determine the type
+	switch data[0] {
+	case '{':
+		var parent jsonParent
+		err := json.Unmarshal(data, &parent)
+		if err != nil {
+			return err
+		}
+		*op = append(*op, parent)
+		return nil
+
+	case '[':
+		// we can't use the jsonParents type or Unmarshal will call back this function
+		var parents []jsonParent
+		err := json.Unmarshal(data, &parents)
+		if err != nil {
+			return err
+		}
+		*op = append(*op, parents...)
```

Comment:
similarly, could this simple be `*op = parents;` ?

Reply:
unfortunately not, it seems that golang doesn't allow you to simply cast a slice of objects to a slice type, even if the underlying object is the same. i.e. you can't just do:
```
*op = jsonParents{parents}
```
I think you could initialise the slice first then use copy:
```
*op = make(jsonParents,sizeof(parents))
copy(*op, parents)
```
but this is basically doing append but in two lines instead of one....